aws_advanced_python_wrapper/blue_green_plugin.py

```diff
@@ -0,0 +1,1913 @@
+                monitor.should_collect_topology.clear()
+                monitor.use_ip_address.set()
+        elif phase == BlueGreenPhase.COMPLETED:
+            for monitor in self._monitors:
+                monitor.interval_rate = BlueGreenIntervalRate.BASELINE
+                monitor.should_collect_ip_addresses.clear()
+                monitor.should_collect_topology.clear()
+                monitor.use_ip_address.clear()
+                monitor.reset_collected_data()
+
+            # Stop monitoring old1 cluster/instance.
+            if not self._rollback and self._monitors[BlueGreenRole.SOURCE.value] is not None:
+                self._monitors[BlueGreenRole.SOURCE.value].stop.set()
+        else:
+            raise UnsupportedOperationError(
+                Messages.get_formatted(
+                    "BlueGreenStatusProvider.UnknownPhase", self._bg_id, self._summary_status.phase))
+
+    def _update_status_cache(self):
+        latest_status = self._plugin_service.get_status(BlueGreenStatus, self._bg_id)
+        self._plugin_service.set_status(BlueGreenStatus, self._summary_status, self._bg_id)
+
+        if latest_status is not None:
+            # Notify all waiting threads that the status has been updated.
+            with latest_status.cv:
+                latest_status.cv.notify_all()
+
+    def _log_current_context(self):
+        logger.debug(f"[bg_id: '{self._bg_id}'] Summary status: \n{self._summary_status}")
+        hosts_str = "\n".join(
+            f"   {blue_host} -> {host_pair[1] if host_pair else None}"
+            for blue_host, host_pair in self._corresponding_hosts.items())
+        logger.debug(f"Corresponding hosts:\n{hosts_str}")
+        phase_times = \
+            "\n".join(f"   {event_desc} -> {info.date_time}" for event_desc, info in self._phase_times_ns.items())
+        logger.debug(f"Phase times:\n{phase_times}")
+        change_name_times = \
+            "\n".join(f"   {host} -> {date_time}" for host, date_time in self._green_host_name_change_times.items())
+        logger.debug(f"Green host certificate change times:\n{change_name_times}")
+        logger.debug("\n"
+                     f"   latest_status_phase: {self._latest_phase}\n"
+                     f"   blue_dns_update_completed: {self._blue_dns_update_completed}\n"
+                     f"   green_dns_removed: {self._green_dns_removed}\n"
+                     f"   all_green_hosts_changed_name: {self._all_green_hosts_changed_name}\n"
+                     f"   green_topology_changed: {self._green_topology_changed}\n")
+
+    def _log_switchover_final_summary(self):
+        switchover_completed = (not self._rollback and self._summary_status.phase == BlueGreenPhase.COMPLETED) or \
+                               (self._rollback and self._summary_status.phase == BlueGreenPhase.CREATED)
+        has_active_switchover_phases = \
+            any(phase_info.phase is not None and phase_info.phase.is_active_switchover_completed
```

Comment:
```suggestion
            any(phase_info.phase is not None and phase_info.phase.is_switchover_active_or_completed()
```

Reply:
The log switchover final summary output's formatting looks off, would you be able to verify it?
```
----------------------------------------------------------------------------------
timestamp                         time offset (ms)                           event----------------------------------------------------------------------------------
>28s                    ms                                
----------------------------------------------------------------------------------
```